File: 코딩 기초 트레이닝/n개 간격의 원소들-p181888/n개 간격의 원소들.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181888

function solution(num_list, n) {
    var answer = [];
    let cnt = 0;

    num_list.forEach((num, index) => {
        if (cnt === 0) {
            answer.push(num);
            cnt = n -1;
        } else {
            cnt -= 1;
        }
    });

    return answer;
}

File: 코딩 기초 트레이닝/n개 간격의 원소들-p181888/n개 간격의 원소들.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181888

def solution(num_list, n):
    answer = []
    cnt = 0

    for index, num in enumerate(num_list):
        if cnt == 0:
            answer.append(num)
            cnt = n-1
        else:
            cnt -= 1

    return answer
